### ISSUE TITLE
Bad practice to publish angular / react app and server app to private NPM repo ?

### ISSUE THREAD

**Issue by appsolutegeek** [lerna/lerna#1268]
Hi,

I thought I almost had it :-) Built a small POC to manage a number of libraries both for client (angular and react) and server (nodejs).

Everything seems to be working but this is under the assumption that it is NOT bad practice to publish client web app and server app using lerna to the private npm repo.

Would anyone care to comment ?

Right now, this is what I have, so I have a number of libraries, which are shared amongst the client and server apps - these libraries are published with LERNA BUT also the client and server apps are also published as a npm package.

On deployment, all I need to is a 

`Yarn add @mycompany/my-web-app`

and because I make use of the bin inside of the package.json, then I am able to automatically run the app using 

`yarn execute-app`

Initially, this seemed a great idea but then I saw some posts saying that we should be ONLY deploying libraries to NPM repos (even though mine is private).

So, I thought I would reach out to see if anybody has other options or ideas ?

Thanks in advance

**Comment by zenflow** [comment]
@appsolutegeek Excellent question! Thank you for bringing up this discussion, as I had similar concern about best practice

> Initially, this seemed a great idea but then I saw some posts saying that we should be ONLY deploying libraries to NPM repos (even though mine is private).

Do you happen to have a reference for this? I recall reading a comment saying that lerna is for __packages__, not apps. But a package can be a library __or an app__, so I don't think they meant that you shouldn't use lerna for apps, just that lerna is not tailored for apps specifically.

I myself don't see any problem with lerna repos containing apps, and it is super convenient to be able to version some apps together, with their common dependencies (library packages).

I look forward to hearing if this practice is not recommended for whatever reason!

**Comment by evocateur** [comment]
I use lerna to develop apps almost every day. It does just fine.

_However_, I do not publish those apps, they merely consume the local lerna packages (which _are_ published, sharing bits of formerly-internal company logic). I have a webpack build that creates the tarball artifact for deployment in various server environments (test/staging/live, etc).

Whether or not you publish the "application" packages to an npm registry is irrelevant in this context, however. The important distinction is that you **do not** use npm/yarn (or lerna) to _deploy_ your code. npm and lerna are not meant for that job, and there emphatically _will not_ be any explicit support for it from lerna.

TL;DR: Your CI server should be creating an artifact that you use to deploy, not lerna/npm directly.

**Comment by appsolutegeek** [comment]
Great, thanks. This now makes total sense.
